libyara/modules/pe.c

```diff
@@ -396,12 +402,20 @@ int64_t pe_rva_to_offset(
         section_offset = section->PointerToRawData;
         section_raw_size = section->SizeOfRawData;
 
-        // If the section_offset is less than 0x200 it is rounded down to 0.
-        // See also: https://github.com/plusvic/yara/issues/399
-        // Discussion (and other awesome details) at:
+        // Round section_offset down to file alignment.
+        //
+        // Rounding everything less than 0x200 to 0 as discussed in
         // https://code.google.com/archive/p/corkami/wikis/PE.wiki#PointerToRawData
-        if (section_offset < 0x200)
-          section_offset = 0;
+        // does not work for PE32_FILE from the test suite and for
+        // some tinype samples where File Alignment = 4
+        // (http://www.phreedom.org/research/tinype/).
+        int alignment = OptionalHeader(pe, FileAlignment);
+        if (alignment) {
```

Comment:
Just a minor style-related change. Opening curling braces for ``if``, ``while``, ``switch``, etc, are put in the next line in all the codebase, please follow that style.